### ISSUE TITLE
Commit churn with update-lockfile

### ISSUE THREAD

**Issue by djmitche** [renovatebot/config-help#458]
### Which Renovate are you using? e.g. CLI, GitHub app, GitLab app, etc.

GH App

### Which platform are you using? GitHub, GitLab, Bitbucket Azure DevOps

GH

### What is your question?

Renovate seems to be "thrashing" between two different updates in https://github.com/taskcluster/taskcluster/pull/2115.  At a guess, this is because there are two different package.json's that it's updating, but both map to the same PR name.  Is there any way to prevent this, or separate the two out into different PRs?

**Comment by rarkins** [comment]
I’m not sire it’s caused by updating two files at once because that’s pretty common. More likely is that the update-lockfile rangeStrategy is causing the churn somehow. I’ll see if the logs give a hint as to why it thinks the branch needs updating.

**Comment by rarkins** [comment]
Logging my troubleshooting here for record.

First thing to note is that the churn is *not* because Renovate thinks versions are changing. The PR body itself has only changed a couple of times, which seems about right:

![image](https://user-images.githubusercontent.com/6311784/70423702-b26a5e80-1a76-11ea-94b2-66b6ccb75864.png)

Next up: is Renovate churning it *every* time?

Renovate has run approximately hourly:

![image](https://user-images.githubusercontent.com/6311784/70423859-01b08f00-1a77-11ea-8bb0-9f29b2eec66f.png)

But it was last updated (churned?) yesterday:
![image](https://user-images.githubusercontent.com/6311784/70423882-0e34e780-1a77-11ea-9819-5a7b2b6c20ea.png)

I looked at the log of when it last changed, and the log immediately before that to compare. It seems that the cause of the `yarn.lock` changing was due to a transitive dependency being updated:

![image](https://user-images.githubusercontent.com/6311784/70424713-7c2dde80-1a78-11ea-8e96-68363a398ba7.png)

This is non-ideal, but also "expected" based on the current approach of Renovate. i.e. it will use `yarn` to update the `yarn.lock`, then do a compare of it to the existing `yarn.lock`. If there's a difference, there will be a force push. This should only happen when a sub-dependency of one of the babel packages updates and not because any random transitive package is updated.
